### DISCUSSION TITLE
Query the status of a [power generic_homeassistant] switch from a macro in Klipper.

### DISCUSSION THREAD

**Discussion by JCBond** [mainsail-crew/mainsail#1839] [Help] [answered]
I have defined a [power generic_homeassistant] switch in my Moonraker.conf file and that shows up correctly in my web UI. I want to be able to query the state of that toggle from an IF statement in a Macro within Klipper.

Does anyone know if it is possible to do this, and if so, what would be the IF statement in the Macro?

**Comment by meteyou** [comment, marked as answer]
Klipper can only check Klipper states. So, pulling the state from a Moonraker object is impossible.
